extensions/windowgram/init.lua

```diff
@@ -0,0 +1,210 @@
+--- === hs.windowgram ===
+---
+--- region generation using ascii rows similar to tmuxomatic for window areas
+
+local windowgram = {}
+
+-- Returns a table with the relative cols and rows.
+function parseWindowgram(wg)
+  local gridw = nil
+
+  local grid = {}
+
+  -- Split into lines
+  local lines = {}
+  for line in string.gmatch(wg, "[^\r\n]+") do
+    if #line > 0 then
+      table.insert(lines, line)
+    end
+  end
+  lines = cleanLines(lines)
+
+  for i, line in ipairs(lines) do
+    if gridw then
+      if gridw ~= line:len() then
+        error('inconsistent grid width in windowgram')
+      end
+    else
+      gridw=line:len()
+    end
+    for column = 1, #line do
+      local char = line:sub(column, column)
+      if not grid[char] then
+        -- new window, create it with size 1x1
+        grid[char] = { x = column , y = i}
+      else
+        -- expand it
+        grid[char].w = column
+        grid[char].h = i
+      end
+    end
+  end
+  return grid
+
+end
+
+function cleanLines(wg)
+  local target = {}
+  for i,l in ipairs(wg) do
+    -- Remove comments and whitespace
+    l = l:gsub('#.*','')
+    l = l:gsub("%s+","")
+    table.insert(target, l)
+  end
+  return target
+end
+
+--- hs.windowgram.getRegionRatios(regions)
+--- Function
+--- Gets a table with the regions as percentages of the screen resolution
+---
+--- Parameters:
+---  * regions - The regions table returned by windowgram.convertToRegions(wg)
```

Comment:
Mentions of other API calls in the docstrings should be formatted like \`hs.windowgram.convertToRegions()\`